### PR TITLE
GetCitiesAverageFruitsAndVegetablesPricesLowestCost

### PR DESCRIPTION
**Pull request by youssefahmed55**
Update GetCitiesAverageFruitsAndVegetablesPricesLowestCost

### CODE REVIEW

File: src/test/kotlin/interactor/GetCitiesAverageFAVLowesCostTest.kt [hunk 1]
```diff
@@ -10,19 +10,56 @@ import org.junit.jupiter.api.TestInstance
...
     fun setUp() {
-        fakeNormalData = FakeDataGetCitiesAverageFAVLowesCost()
+        fakeDataGetCitiesAverageFAVLowesCostThatCitiesHaveNotNullSalariesOrNullFruitsAndVegetablesPrices = FakeDataGetCitiesAverageFAVLowesCostThatCitiesHaveNotNullSalariesOrNullFruitsAndVegetablesPrices()
+        fakeDataGetCitiesAverageFAVLowesCostThatSomeCitiesHaveNullSalariesOrNullFruitsAndVegetablesPrices = FakeDataGetCitiesAverageFAVLowesCostThatSomeCitiesHaveNullSalariesOrNullFruitsAndVegetablesPrices()
+        fakeDataAllCitiesHaveNullSalaryCase = FakeDataAllCitiesHaveNullSalaryCase()
+        fakeDataAllCitiesHaveNullPricesOfFruitsAndVegetables =  FakeDataAllCitiesHaveNullPricesOfFruitsAndVegetables()
     }
     @Test
-    fun should_ReturnTenCitiesNames_When_GetAverageOfFruitsAndVegetablesLowestCost() {
+    fun should_ReturnAtLeastTenCitiesNames_When_GetCitiesNamesThatAverageOfFruitsAndVegetablesHaveLowestCostAndAllCitiesHaveNotNullSalariesOrNullFruitsAndVegetablesPrices() {
```
Comment (Abdelrahman-Sedky) [inline src/test/kotlin/interactor/GetCitiesAverageFAVLowesCostTest.kt:25]: **Please reconsider the following:**
1. The naming does not clarify the test case.
2. You can make the name short.
>  Ex: `should return at most ten cities names when given data with not null salaries and not null fruit and vegetable prices` 
3. use `` in naming for better readability.

**Review by Abdelrahman-Sedky — CHANGES_REQUESTED** [review]
The Logic LGTM.
Some changes is required in testing.

**Review by Abdelrahman-Sedky — APPROVED** [review]
LGTM

File: src/main/kotlin/interactor/GetCitiesAverageFAVLowesCost.kt [hunk 2]
```diff
@@ -10,7 +10,7 @@ class GetCitiesAverageFAVLowesCost (
     fun execute(limit: Int): List<String> {
         return dataSource
             .getAllCitiesData()
-            .filter(::excludeNullSalariesAndFAVPrices)
+            .filter(::isNotNullSalaryAndAnyOfFruitsAndVegetablesPricesAreNotNull)
             .sortedByDescending {getAverageFAVPricesCompareToSalaries(it.fruitAndVegetablesPrices,it.averageMonthlyNetSalaryAfterTax!!)}
             .take(limit)
             .map { it.cityName }
```
Comment (Abdelwahab-IO) [inline src/main/kotlin/interactor/GetCitiesAverageFAVLowesCost.kt:16]: use scope functions
handle negative limit case

File: src/test/kotlin/interactor/GetCitiesAverageFAVLowesCostTest.kt [hunk 3]
```diff
@@ -1,6 +1,6 @@
 package interactor
```
Comment (Abdelwahab-IO) [inline src/test/kotlin/interactor/GetCitiesAverageFAVLowesCostTest.kt:2]: add test case to handle limit with negative numbers

**Review by Abdelwahab-IO — CHANGES_REQUESTED** [review]

**Review by Abdelwahab-IO — APPROVED** [review]
